esphome/components/light/effects.py

```diff
@@ -408,7 +414,8 @@ async def addressable_color_wipe_effect_to_code(config, effect_id):
                 ("b", int(round(color[CONF_BLUE] * 255))),
                 ("w", int(round(color[CONF_WHITE] * 255))),
                 ("random", color[CONF_RANDOM]),
-                ("num_leds", color[CONF_NUM_LEDS]),
+                ("num_leds", num_leds),
+                ("gradient", gradient),
```

Comment:
After adding defaults to the validation, remove the `num_leds` and `gradient` variables above and just use:
```suggestion
                ("num_leds", config[CONF_NUM_LEDS]),
                ("gradient", config[CONF_GRADIENT]),
```

Reply:
fixed.